src/pages/SearchPage/Heatmap/Heatmap.js

```diff
@@ -37,6 +38,17 @@ const hoursList = [
 ];
 
 function Heatmap({ info }) {
+  const [selected, setSelected] = useState({});
+  function showPostsTable() {
```

Comment:
I don't think the try-catch clause is best practice here. You can better write something like this instead of the `showPostsTable` function

```
const selectedPosts = selected.day !== undefined ? info[selected.day][selected.hour] : [];

...

{selectedPosts.length > 0 && <PostsTable info={info[selected.day][selected.hour]} />}
```